code/controllers/CMSMain.php

```diff
@@ -88,8 +97,24 @@ public function init() {
 
 		CMSBatchActionHandler::register('publish', 'CMSBatchAction_Publish');
 		CMSBatchActionHandler::register('unpublish', 'CMSBatchAction_Unpublish');
-		CMSBatchActionHandler::register('delete', 'CMSBatchAction_Delete');
-		CMSBatchActionHandler::register('deletefromlive', 'CMSBatchAction_DeleteFromLive');
+		
+
+		// Check legacy actions
+		$legacy = $this->config()->enabled_legacy_actions;
```

Comment:
See the docs on https://github.com/silverstripe/silverstripe-framework/pull/4189; This is a config to prevent the core api (CMSMain) from automatically calling register on any of the batch handlers. It provides a standardised way to control which deprecated actions should be enabled.

See SiteTree.php for the other place this is used; If we enable the old delete mechanism, then the "archive" popup action will also be replaced with the old "delete" action in order to be consistent.

This is just a temp API until 4.0, where we will remove it, so it's only necessary for the 3.x branch onwards.